cli/spec.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2014,2015,2016,2017 Docker, Inc.
+// Copyright (c) 2018 Huawei Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
Ah, I thought the standard for this project (the whole of kata) was the SPDX style copyright like:
https://github.com/kata-containers/runtime/blob/master/cli/fatal.go#L3
but, I note we have some non-SPDX copyright in the code base already, such as:
https://github.com/kata-containers/runtime/blob/master/cli/config.go#L3

@egernst @sboeuf @jodh-intel , thoughts?
And, maybe we can add a static check to catch any future deviations (whichever way we go) to the CI - maybe.

Other than that, this copyright looks correct to me afaict - the original code coming from https://github.com/opencontainers/runc/blob/master/spec.go I believe. That original file does not have an explicit copyright (it relies on the project wide LICENSE file I believe), so I presume we looked at the git logs to get the date range in the above - which I believe is fine and correct.

Maybe we can add a URL reference to that original file just below the copyright to make the historical origins clear?

Reply:
btw, if we do all agree on SPDX (which we did I think at project inception, but we probably did not document well enough, and have not always caught transgressions), then I'm happy to Issue/PR to update/fix them all etc.

Reply:
Good spot @grahamwhaley! Yep, I agree - we should ensure all our files have the SPDX header where possible.

I've raised https://github.com/kata-containers/tests/issues/240 to add a CI check to enforce this for new / modified files, but would be great to have the existing codebase "SPDX-ified" :smile: